__tests__/atomWithLocation_spec.tsx

```diff
@@ -114,6 +114,75 @@ describe('atomWithLocation', () => {
     expect(window.location.pathname).toEqual('/1');
     expect(window.history.length).toEqual(3);
   });
+
+  it('can override atomOptions', async () => {
+    const locationAtom = atomWithLocation({ replace: false });
+
+    const Navigation = () => {
+      const [location, setLocation] = useAtom(locationAtom);
+      return (
+        <>
+          <div> current pathname in atomWithLocation: {location.pathname} </div>
+          <button type="button" onClick={() => window.history.back()}>
+            back
+          </button>
+          <button type="button" onClick={() => setLocation({ pathname: '/1' })}>
+            button1
+          </button>
+          <button
+            type="button"
+            onClick={() =>
+              setLocation(
+                {
+                  pathname: '/2',
+                },
+                // @ts-expect-error: TODO: How can we fix this error? Expected 1 arguments, but got 2
+                { replace: true },
+              )
+            }
+          >
+            button2
+          </button>
+        </>
+      );
+    };
+
+    const { findByText, getByText } = render(
+      <StrictMode>
+        <Navigation />
+      </StrictMode>,
+    );
+
+    const previousTestHistoryLength = 3;
+
+    await findByText('current pathname in atomWithLocation: /');
+    expect(window.location.pathname).toEqual('/');
+    expect(window.history.length).toEqual(previousTestHistoryLength);
+
+    await userEvent.click(getByText('button1')); // TODO: Why doesn't it increment the history length?
```

Comment:
For some unknown reason history length doesn't increment when clicking it once here.

But it increments if you click it two times.